src/sparsezoo/v2/objects/model_objects.py

```diff
@@ -114,3 +119,73 @@ def _validate_model(self, model: onnx.ModelProto) -> bool:
                         f"Found keys in numpy {key_type}: {list(numpy_dict.keys())}."
                     )
         return True
+
+
+class SelectDirectory(Directory):
+    """
+    Object that represents a directory from which the
+    user can fetch the contained directories and file
+    by key values.
+
+    :param files: list of files contained within the SelectDirectory
+    :param name: name of the NumpyDirectory
+    :param path: path of the NumpyDirectory
+    :param url: url of the NumpyDirectory
+    :param owner_path: path of the parent SelectDirectory
+    """
+
+    def __init__(
+        self,
+        files: List[File],
+        name: str,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+        owner_path: Optional[str] = None,
+    ):
+        self._default, self._available = None, None
+
+        super().__init__(
+            files=files, name=name, path=path, url=url, owner_path=owner_path
+        )
+
+        self.files_dict = self.files_to_dictionary()
+
+    def __getitem__(self, key):
+        file = self.files_dict[key]
+        file.get_path()
+        return file
+
+    def files_to_dictionary(self):
```

Comment:
Recommend just putting this code directly in `__init__` if its not called anywhere else (doesn't seem like it is?)

Reply:
Very good idea.

Reply:
On the other hand, I like the aesthetics of not overloading the constructor with lines of code. Keeping it as a method, even though it's only called once, is just more readable I think.